main.c

```diff
@@ -407,12 +408,17 @@ static void update_info(void)
 	l->p = l->buf;
 	r->p = r->buf;
 	if (mode == MODE_THUMB) {
-		if (tns.loadnext < tns.end)
+		if (tns.loadnext < tns.end) {
 			bar_put(l, "Loading... %0*d", fw, tns.loadnext + 1);
-		else if (tns.initnext < filecnt)
+		} else if (tns.initnext < filecnt) {
 			bar_put(l, "Caching... %0*d", fw, tns.initnext + 1);
-		else
-			strncpy(l->buf, files[fileidx].name, l->size);
+		} else {
+			const char *filename = files[fileidx].name;
```

Comment:
`files.name` is given by the user, *as is*. So this won't work if the user gave relative path.

```console
$ cd $HOME
$ nsxiv ../user/pic/a.png
```
in this case, `.name` will be `../user/pic/a.png`

Reply:
..that is intended. I don't see why it would make sense to shorten the absolute path in the case where the user hasn't given an absolute path. That would be confusing IMO.